File: csharp/hanoi_tower.cs
namespace ConsoleApp1
{
    class Program {
      static long hanoi(int n, int from, int mid, int to) {
        if(n == 1) {
          return 1;
        }
        return hanoi(n-1, from, to, mid) + 1 + hanoi(n-1, mid, from, to);
      }

      static void Main(string[] args) {
        System.Console.WriteLine(hanoi(32, 1, 2, 3));
      }
    }
}

File: scala/hanoi-tower.scala
def hanoi(n: Long, a: String, b: String, c: String): Long =
    if (n == 1) then 1
    else hanoi(n-1, a, c, b) + 1 + hanoi(n-1, b, a, c)


def main(args: Array[String]): Unit = println(hanoi(32, "A", "B", "C"))